File: solutions/965. Univalued Binary Tree/965.py
class Solution:
  def isUnivalTree(self, root: Optional[TreeNode]) -> bool:
    if not root:
      return True
    if root.left and root.left.val != root.val:
      return False
    if root.right and root.right.val != root.val:
      return False
    return self.isUnivalTree(root.left) and self.isUnivalTree(root.right)

File: solutions/965. Univalued Binary Tree/965.java
class Solution {
  public boolean isUnivalTree(TreeNode root) {
    if (root == null)
      return true;
    if (root.left != null && root.left.val != root.val)
      return false;
    if (root.right != null && root.right.val != root.val)
      return false;
    return isUnivalTree(root.left) && isUnivalTree(root.right);
  }
}
